src/utils/npm.js

```diff
@@ -1,18 +1,50 @@
 // @flow
-import * as processes from './processes';
+import {PError} from './errors';
 import * as logger from './logger';
+import * as processes from './processes';
 import pLimit from 'p-limit';
 
-const npmRequestLimit = pLimit(20);
+const npmRequestLimit = pLimit(40);
 
 export function info(pkgName: string) {
   return npmRequestLimit(async () => {
     logger.info(`npm info ${pkgName}`);
 
-    let result = await processes.spawn('npm', ['info', pkgName, '--json'], {
+    const result = await processes.spawn('npm', ['info', pkgName, '--json'], {
       silent: true,
     });
 
     return JSON.parse(result.stdout);
   });
 }
+
+export function addTag(pkgName: string, pkgVersion: string, tag: string) {
+  return npmRequestLimit(async () => {
+    const pkgStr = `${pkgName}@${pkgVersion}`;
+    logger.info(`npm dist-tag ${pkgStr} ${tag}`);
+
+    const result = await processes.spawn('npm', ['dist-tag', 'add', pkgStr, tag]);
```

Comment:
There's an interesting case here, not sure if we need to worry about it or not.
If a tag already exists but at a different version of the package, this command wont throw any errors.

It's *probably* not a huge issue, as you'd have to be behind master for this to ever occur (or someone else has manually published and that change wasnt in master).

The alternative would be to do an `npm dist-tag ls` but that just opens up more chance for race conditions, so probably not important.

Reply:
As discussed, will add an npm-info check here as an extra precaution.

Reply:
Ended up adding this check to locks.lock()